src/compiler/emitter.ts

```diff
@@ -2269,7 +2269,13 @@ namespace ts {
                 writePunctuation(".");
             }
 
-            emitTokenWithComment(token.kind, node.expression.end, writePunctuation, node);
+            if (!node.questionDotToken) {

```

Comment:
I think @rbuckton should review this.. There is `shouldEmitDotDot` just above this. Seems like, all of that logic should be combined together..

Reply:
Since we synthesize the `.` token above, we could possibly just change this to `emit(token)` and remove the `if`? That might impact comment emit, however. 

@dragomirtitian: If you could make this change, we can investigate any test failures that might result from the difference in comment emit and possibly modify `getDotOrQuestionDotToken` to address differences resulting from calling `emit` rather than `emitTokenWithComment`.

Reply:
@rbuckton I had a look at using `emit` and the fallout in the general emit is pretty extensive and undesirable. It generally involves emitting extra trivia from either the `expression` or  the `name` side. I'll try to fix it tomorrow, as for me it's pretty late now.

Reply:
@rbuckton  As suggested I use `emit` now and make the necessary position adjustments in `getDotOrQuestionDotToken`